public/app/features/explore/ReturnToDashboardButton.tsx

```diff
@@ -11,15 +11,30 @@ import { setDashboardQueriesToUpdateOnLoad } from '../dashboard/state/reducers';
 import { isSplit } from './state/selectors';
 import { locationService } from '@grafana/runtime';
 
-interface Props {
-  exploreId: ExploreId;
-  splitted: boolean;
-  queries: DataQuery[];
-  originPanelId?: number | null;
-  setDashboardQueriesToUpdateOnLoad: typeof setDashboardQueriesToUpdateOnLoad;
+function mapStateToProps(state: StoreState, { exploreId }: { exploreId: ExploreId }) {
+  const explore = state.explore;
+  const splitted = isSplit(state);
+  const { datasourceInstance, queries, originPanelId } = explore[exploreId]!;
+
+  return {
+    exploreId,
+    datasourceInstance,
+    queries,
+    originPanelId,
+    splitted,
+  };
 }
 
-export const UnconnectedReturnToDashboardButton: FC<Props> = ({
+const mapDispatchToProps = {
+  setDashboardQueriesToUpdateOnLoad,
+};
+
+const connector = connect(mapStateToProps, mapDispatchToProps);
+type PropsFromRedux = ConnectedProps<typeof connector>;
+
+interface ReturnToDashboardButtonProps extends PropsFromRedux {}
+
+export const UnconnectedReturnToDashboardButton: FC<ReturnToDashboardButtonProps> = ({
```

Comment:
`interface ReturnToDashboardButtonProps extends PropsFromRedux {}` is identical to `PropsFromRedux`, I think in this case we can skip it and just do:

```suggestion
const connector = connect(mapStateToProps, mapDispatchToProps);
type Props = ConnectedProps<typeof connector>;

export const UnconnectedReturnToDashboardButton: FC<Props> = ({
```

What do you think?
cc. @zoltanbedi I'm not sure if we really have any preference on this, I don't think we disallow empty interfaces in our eslint config. any opinion on this one?

Reply:
Yeah that's true! Didn't know if you wanted it removed or to have it there for the sake of adding props in the future. But i'll remove it then

Reply:
Oh i had no idea you could add the commit from here haha. I went ahead and did it manually xD

Reply:
Agree with Gio, as far as I know there is no hard preference here. My personal is to avoid having empty interfaces.